hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/KeyGenUtils.java

```diff
@@ -153,7 +154,12 @@ public static String getRecordKey(GenericRecord record, List<String> recordKeyFi
     StringBuilder recordKey = new StringBuilder();
     for (int i = 0; i < recordKeyFields.size(); i++) {
       String recordKeyField = recordKeyFields.get(i);
-      String recordKeyValue = HoodieAvroUtils.getNestedFieldValAsString(record, recordKeyField, true, consistentLogicalTimestampEnabled);
+      String recordKeyValue;
+      try {
+        recordKeyValue = HoodieAvroUtils.getNestedFieldValAsString(record, recordKeyField, false, consistentLogicalTimestampEnabled);
```

Comment:
Could you add a test case where the nested field does not exist and assert that the exception is thrown?

Reply:
Also create a follow-up JIRA to revisit all places this method can return null for non-existent field.

Reply:
https://issues.apache.org/jira/browse/HUDI-8256